orangecontrib/text/corpus.py

```diff
@@ -100,6 +100,30 @@ def extend_corpus(self, metadata, Y):
         self.X = self.W = np.zeros((len(self), 0))
         Table._init_ids(self)
 
+    def update_attributes(self, X, dictionary):
+        """
+        Adds X to the corpus and updates its domain accordingly.
+        """
+        self.X = X
+
+        features = [
+            dictionary[key]
+            for key
+            in range(len(dictionary.keys()))
```

Comment:
What if keys wouldn't be compact, as it can happen if you filter out extremes. I think `sorted(dictionary.keys())` is safer.